### PR TITLE
Replica pods can show the blood DNA of injected samples

### PR DESCRIPTION
**Pull request by 81Denton**
## About The Pull Request
Using the plant analyzer on a replica pod shows the blood DNA of any blood samples it contains.

## Why It's Good For The Game
With the blood DNA, people can figure out whose blood is actually inside the pod. It doesn't straight up show the name, so it'll require a bit of effort to figure out (and the medical records/a forensics scanner).

## Changelog
:cl: Denton
tweak: Plant analyzers can now scan replica pods for the blood DNA (also known as UE) of any blood samples inside. The blood DNA is listed in the medical records and verifiable with forensics scanners.
/:cl:

### CODE REVIEW

**Comment by TheChosenEvilOne** [comment]
could it be possible to make it inform the ghost when it is planted?

**Comment by JMoldy** [comment]
Wait, does the clone pod not still revive the person whose blood was injected?

**Comment by 81Denton** [comment]
@TheChosenEvilOne That's definitely possible, but I'd prefer to do it in another PR.
@JMoldy How do you mean?

**Comment by Iatots** [comment]
Maybe they are asking why does it need to notify the podcloned person instead of just shoving the ghost in the mob like cloning did?

**Comment by 81Denton** [comment]
Does it currently force move the ghost into the new body? I'm having a hard time testing it due to firewall issues atm.

**Comment by ExcessiveUseOfCobblestone** [comment]
it forced the mind into it if it's a ghost

**Comment by 81Denton** [comment]
Alright, I'll remove the notify part once I'm back home.

File: code/modules/hydroponics/grown/replicapod.dm [hunk 1]
```diff
@@ -119,6 +121,7 @@
 		podman.hardset_dna(null,null,podman.real_name,blood_type, new /datum/species/pod,features)//Discard SE's and UI's, podman cloning is inaccurate, and always make them a podman
 		podman.set_cloned_appearance()
 		log_cloning("[key_name(mind)] cloned as a podman via [src] in [parent] at [AREACOORD(parent)].")
+		podman.notify_ghost_cloning("You have been cloned by a replica pod and can now enter your new body.", source = podman)
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/replicapod.dm]: imo, the message should be sent when the seed is PLANTED, not when it's HARVESTED

we're already immediately putting the ghost in their new podperson body upon the harvesting of the seed anyway, right? why do we need to send them a message about it?

**Comment by JMoldy** [comment]
My bad, I was confused. I thought this meant that it would alert all ghosts in the way a ghostrole spawn does for anyone to be able to take. 
As it is podclones already force ghosts into the podperson when harvested, they just should be alerted when _planted_ to be consistent.

**Comment by 81Denton** [comment]
I agree, it would be good if there was an alert when planted. I'll look into it for another PR.

File: code/modules/hydroponics/grown/replicapod.dm [hunk 2]
```diff
@@ -22,6 +22,7 @@
 	var/list/features = null
 	var/factions = null
 	var/list/quirks = null
+	var/sampleDNA = null
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/replicapod.dm]: can you delete this garbo so it's all

`var/XYZ` instead of `var/XYZ = null`

Comment (81Denton) [reply code/modules/hydroponics/grown/replicapod.dm]: yeah that's reasonable

**Pull request merged**